test/e2e/create_test.go

```diff
@@ -160,30 +160,63 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
-	// provisioned and then change the key. The correct amount of nodes being configured is proof that the
-	// mismatched Machine created with the mismatched key was deleted and replaced.
-	// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-	// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
-	// Azure.
+
+	t.Run("Machine configuration while private key change", tc.testMachineConfigurationWhilePrivateKeyChange)
+
 	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
 	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-	if tc.CloudProvider.GetType() == config.AzurePlatformType {
-		// Replace the known private key with a randomly generated one.
-		err = tc.createPrivateKeySecret(false)
-		require.NoError(t, err, "error replacing private key secret")
-	}
 	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
 	assert.NoError(t, err, "Windows node creation failed")
-	if tc.CloudProvider.GetType() == config.AzurePlatformType {
-		// Update the machines list
-		machines, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Running", false)
-		require.NoError(t, err, "error waiting for Windows Machines to be running")
-	}
 	tc.machineLogCollection(machines.Items)
 }
 
+// testMachineConfigurationWhilePrivateKeyChange tests that machines which have not yet been configured by WMCO are
+// deleted after the private key is changed, but before WMCO is able to configure them, resulting in WMCO getting an
+// SSH authentication error. This could be considered a platform-agnostic test (except for vSphere where the private
+// key is baked in the VM template) so we run it only on Azure.
+func (tc *testContext) testMachineConfigurationWhilePrivateKeyChange(t *testing.T) {
+	if tc.CloudProvider.GetType() != config.AzurePlatformType {
+		t.Skip("test disabled, exclusively runs on Azure")
+	}
+	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
+	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
+
+	err = tc.createPrivateKeySecret(false)
+	require.NoError(t, err, "error replacing private key secret")
+
+	err = tc.waitForMachinesDeleted(machines.Items)
+	require.NoError(t, err, "error waiting for machines deletion after private key secret change")
+}
+
+// waitForMachinesDeleted waits for the given list of machines to be deleted
+func (tc *testContext) waitForMachinesDeleted(machines []mapi.Machine) (err error) {
+	// This is the maximum amount of time for the deletion of all machines in Azure
+	deletionTimeout := time.Minute * 15
+	for _, m := range machines {
+		log.Printf("waiting (timeout: %s) for machine %s to be deleted", deletionTimeout.String(), m.GetName())
+		err = wait.Poll(retry.ResourceChangeTimeout, deletionTimeout, func() (done bool, err error) {
```

Comment:
i wish I caught this in the original review, we shouldn't wait 2 min between polls.

Reply:
See https://github.com/openshift/windows-machine-config-operator/pull/1821#discussion_r1334405568

Reply:
Yeah the intervals defined in that package are 5 and 15 seconds:
https://github.com/openshift/windows-machine-config-operator/blob/833c1b6b6c24c02de9ac68e7f903d092b8b889aa/pkg/retry/retry.go#L8-L11

Reply:
I misunderstood your comment and placed the focus on `ResourceChangeTimeout` after reading the [description](https://github.com/openshift/windows-machine-config-operator/blob/833c1b6b6c24c02de9ac68e7f903d092b8b889aa/pkg/retry/retry.go#L14C27-L14C65)